core/network/src/protocol/sync.rs

```diff
@@ -815,12 +815,7 @@ impl<B: BlockT> ChainSync<B> {
 	/// with or without errors.
 	pub fn on_justification_import(&mut self, hash: B::Hash, number: NumberFor<B>, success: bool) {
 		let finalization_result = if success { Ok((hash, number)) } else { Err(()) };
-		if !self.extra_justifications.try_finalize_root((hash, number), finalization_result, true) {
-			debug!(target: "sync", "Got justification import result for unknown justification {:?} {:?} request.",
-				hash,
-				number,
-			)
-		}
+		self.extra_justifications.try_finalize_root((hash, number), finalization_result, true);
```

Comment:
This can happen if we process the notification for `on_block_finalized` before we actually process the import result. Either way it is fine because we also call `try_finalize_root` on `on_block_finalized` so any necessary bookkeeping is done correctly.

Reply:
@andresilva can you add this to the code?

Reply:
@andresilva I thought [this check](https://github.com/paritytech/substrate/blob/master/core/network/src/protocol/sync/extra_requests.rs#L176) prevents it from happening, no?